sycl/tools/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
```

Comment:
Looks like unrelated change, I suggest to submit it as separate PR

Reply:
The fix was made for use of ```std::enable_if_t``` (since C++ 14) on early steps of implementation of sycl::bit_cast. After that I replaced ```std::enable_if_t``` with ```static_assert``` as it's more user-friendly in compilation logs to find an error.
Finally, as ```static_assert``` is used, revered this.

Reply:
FYI, for `enable_if_t` we have `details::enable_if_t`, which works in C++11 as well

Reply:
Thanks, I found it then I had implemented this, but in my opinion, we should use C++14 features directly as we moved to C++14 for DPC++ runtime.

Reply:
I think that default standard which is used for compiling user apps is a bit different from standard used for our library: someone might still want to use C++11 for some reasons and it would be good to be flexible in this regard.

I'm not entirely sure that this is still actual for C++11, but for example for C++17 features I would strongly prefer that their usage is optional in our library